10_datastructures_lab/01.py

```diff
@@ -1,10 +1,83 @@
-"""
-Write a program that takes two strings
-from the user and checks if they represent
-a valid user name.
-Valid users and passwords:
-    apple => red
-    lettuce => green
-    lemon => yellow
-    orange => orange
-"""
+#!/usr/bin/env python
+
+"""
+This program parse the given hosts file and return the IP address of a given computers list
+
+"""
+__author__ = 'Eabel'
+
+import sys
+import os
+import argparse
+import socket
+from collections import defaultdict
+
+
+def is_valid_ipv4_address(address):
+    try:
+        socket.inet_pton(socket.AF_INET, address)
+    except AttributeError:  # no inet_pton here, sorry
+        try:
+            socket.inet_aton(address)
+        except socket.error:
+            return False
+        return address.count('.') == 3
+    except socket.error:  # not a valid address
+        return False
+
+    return True
+
+def is_valid_ipv6_address(address):
+    try:
+        socket.inet_pton(socket.AF_INET6, address)
+    except socket.error:  # not a valid address
+        return False
+    return True
+
+def is_valid_ip_address(address):
+    return is_valid_ipv4_address or is_valid_ipv6_address(address)
+
+
+def read_hosts_file(file_path):
+    try:
+        f = open(file_path)
+        host_dict = defaultdict(str)
+        for line in f:
+            items = line.split('=')

```

Comment:
That check should also use exceptions: If your `is_valid_ip_address` throws instead of returning false, you can use it as a single line in that loop:

```
for line in f:
    line = line.rstrip()
    host, ip = line.split('=')
    verify_address(ip)
    host_dict[host] = ip
```

Then you can decide in the except block if you want to `continue` or `exit`, and main code is kept concise